contracts/trading/seaport16/zones/immutable-signed-zone/v3/ImmutableSignedZoneV3.sol

```diff
@@ -0,0 +1,834 @@
+        schemas[0].metadata = abi.encode(
+            _domainSeparator(),
+            _apiEndpoint,
+            _getSupportedSubstandards(),
+            _documentationURI
+        );
+    }
+
+    /**
+     * @notice Returns signing information about the zone.
+     *
+     * @return domainSeparator  The domain separator used for signing.
+     * @return apiEndpoint      The API endpoint to get signatures for orders.
+     * @return substandards     The supported substandards.
+     * @return documentationURI The documentation URI.
+     */
+    function sip7Information()
+        external
+        view
+        override
+        returns (
+            bytes32 domainSeparator,
+            string memory apiEndpoint,
+            uint256[] memory substandards,
+            string memory documentationURI
+        )
+    {
+        domainSeparator = _domainSeparator();
+        apiEndpoint = _apiEndpoint;
+
+        substandards = _getSupportedSubstandards();
+
+        documentationURI = _documentationURI;
+    }
+
+    /**
+     * @notice Check if a given order including extraData is currently valid.
+     *
+     * @dev This function is called by Seaport whenever any extraData is
+     *      provided by the caller, before tokens have been transferred.
+     *
+     * @param zoneParameters             The zone parameters containing data related to
+     *                                   the fulfilment execution.
+     * @return authorizedOrderMagicValue A magic value indicating if the order
+     *                                   is currently valid.
+     */
+    function authorizeOrder(
+        ZoneParameters calldata zoneParameters
+    ) external override returns (bytes4 authorizedOrderMagicValue) {
+        // Put the extraData and orderHash on the stack for cheaper access.
+        bytes calldata extraData = zoneParameters.extraData;
+        bytes32 orderHash = zoneParameters.orderHash;
+
+        // Revert with an error if the extraData is empty.
+        if (extraData.length == 0) {
+            revert InvalidExtraData("extraData is empty", orderHash);
+        }
+
+        // We expect the extraData to conform with SIP-6 as well as SIP-7
+        // Therefore all SIP-7 related data is offset by one byte
+        // SIP-7 specifically requires SIP-6 as a prerequisite.
+
+        // Revert with an error if the extraData does not have valid length.
+        if (extraData.length < 93) {
+            revert InvalidExtraData("extraData length must be at least 93 bytes", orderHash);
+        }
+
+        // Revert if SIP-6 version is not accepted (0).
+        if (uint8(extraData[0]) != _ACCEPTED_SIP6_VERSION) {
+            revert UnsupportedExtraDataVersion(uint8(extraData[0]));
+        }
+
+        // extraData bytes 1-21: expected fulfiller.
+        // (zero address means not restricted).
+        address expectedFulfiller = address(bytes20(extraData[1:21]));
+
+        // extraData bytes 21-29: expiration timestamp.
```

Comment:
The doc message is slightly wrong: "extraData bytes 21-28: expiration timestamp."